sources/light_client.move

```diff
@@ -104,57 +112,109 @@ public fun new_light_client(params: Params, start_height: u64, start_headers: ve
 // Helper function to initialize new light client.
 // network: 0 = mainnet, 1 = testnet
 public fun new_btc_light_client(
-    network: u8, start_height: u64, start_headers: vector<vector<u8>>, ctx: &mut TxContext
+    network: u8, start_height: u64, start_headers: vector<vector<u8>>, pre_start_chain_work: u256, ctx: &mut TxContext
 )  {
     let params = match (network) {
         0 => mainnet_params(),
         1 => testnet_params(),
         _ => regtest_params()
     };
-    let lc = new_light_client(params, start_height, start_headers, ctx);
+    let lc = new_light_client(params, start_height, start_headers, pre_start_chain_work, ctx);
     transfer::share_object(lc);
 }
 
 // === Entry methods ===
-
 /// insert new header to bitcoin spv
-public entry fun insert_header(c: &mut LightClient, raw_header: vector<u8>, ctx: &mut TxContext) {
-    let next_header = new_block_header(raw_header);
-    let current_block = c.latest_finalized_block();
+public(package) fun insert_header(c: &mut LightClient, current_block_hash: vector<u8>, next_header: BlockHeader): vector<u8> {
+    let current_block = c.get_light_block_by_hash(current_block_hash);
     let current_header = current_block.header();
 
     // verify new header
     assert!(current_header.block_hash() == next_header.prev_block(), EBlockHashNotMatch);
     let next_block_difficulty = calc_next_required_difficulty(c, current_block, 0);
     assert!(next_block_difficulty == next_header.bits(), EDifficultyNotMatch);
 
+
     // https://learnmeabitcoin.com/technical/block/time
     // we only check the case "A timestamp greater than the median time of the last 11 blocks".
     // because  network adjusted time requires a miners local time.
-    let median_time = calc_past_median_time(c, current_block);
-    assert!(next_header.timestamp() > median_time, ETimeTooOld);
+    let median_time = c.calc_past_median_time(current_block);
+    assert!(next_header.timestamp() >= median_time, ETimeTooOld);
```

Comment:
```suggestion
    assert!(next_header.timestamp() > median_time, ETimeTooOld);
```

Reply:
In https://learnmeabitcoin.com/technical/block/time/ it says "... greater than the median time".